### PR TITLE
Category dto entity

### PR DESCRIPTION
**Pull request by Subhojitgh**
This contains yaml file And Category,Dto classes.

### CODE REVIEW

File: Common/yaml/category.yaml [hunk 1]
```diff
@@ -0,0 +1,353 @@
+openapi: 3.0.3
+info:
+  title: Smart Contact Manager - OpenAPI 3.0
+  description: 
+              This Project was about creatig smart contact manager.
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: Correct the spell of creating

File: Common/yaml/category.yaml [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+              Which can be specifically used for business model.This is web 
+              application which can be used as a simple plugging to any 
+              exixting project.|-
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: Remove the unnecesary pipe and hyphen symbol

File: Common/yaml/category.yaml [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+   
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: subhojitgh1999@gmail.com
+  license:
+    name: Contact Manager
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: It should be Smart Contact Manager

File: Common/yaml/category.yaml [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+                schema:
+                  type: string
+                  example:
+                    messages: Something went wrong
+      get:
+       tags:
+         - Categories
+       summary: Fetch list of all categories
+       description: All the categories will be fetched
+       operationId: findPetsByStatus
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: Why is the operationId findPets*** ?

File: Common/yaml/category.yaml [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                    $ref: '#/components/schemas/Categories'
+                example:
+                    - id: 100
+                      CategoryName: Hr
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: Field names should be camel case

File: Common/yaml/category.yaml [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+              schema:
+                type: string
+                example:
+                  messages: "Something went wrong"
+                  
+  
+   
+     
+  /v1/categories/{categoryId}/contacts:
+      put:
```
Comment (babugarnaik) [inline Common/yaml/category.yaml:109]: add operationId to PUT request

File: Common/yaml/category.yaml [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+                      
+        requestBody:
+         content:
+          application/json:
+            schema:
+              type: array
+              items: 
+                  $ref: '#/components/schemas/ContactId'
+              example: 
+                contatid:
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: Check the spell of contactId

File: Common/yaml/category.yaml [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+                schema:
+                  type: string
+                  example:
+                    messages: Something went wrong
+      get:
+          tags:
+            - Categories
+          summary: Get contacts details by CategoryId
+          description: Fetch the Contacts by CategoryId
+          operationId: getdeatilsbyId
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: Make it camelcase

File: Common/yaml/category.yaml [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+              content:
+                application/json:
+                  schema:
+                    type: array
+                    items:
+                        $ref: '#/components/schemas/Contacts'
+                    example:
+                        - id: 100
+                          fullName: bablu kr
+                          contactcNumber: 7546930695
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: Correct the spell of contact

File: Common/yaml/category.yaml [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+              content:
+                application/json:
+                  schema:
+                    type: string
+                    example:
+                      messages: Something went wrong
+      delete:
+          tags:
+            - Categories
+          summary: Delete Conatacts detais by categoryId
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: Correct the spell of contacts

File: Common/yaml/category.yaml [hunk 11]
```diff
@@ -0,0 +1,353 @@
...
+            - name: Delete
+              in: header
+              description: ' Delete specfic Contact with respect to given CategoryId'
+              required: false
+              schema:
+                type: string
+            
+          responses:
+            '200':
+              description: SUCCESFULLY DELETED
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: Correct the spell of Successfully

File: Common/yaml/category.yaml [hunk 12]
```diff
@@ -0,0 +1,353 @@
...
+        phone:
+          type: string
+          example: '12345'
+        userStatus:
+          type: integer
+          description: User Status
+          format: int32
+          example: 1
+     
+
```
Comment (babugarnaik) [inline Common/yaml/category.yaml]: Remove extra lines

File: smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+public class CategoryRequestDto {
+	
+	@Size(min=2,message="Name should be atleast 2 letters")
+	private String name;
+	
+	@Size(min=2,message="Name should be atleast 2 letters")
+	private String type;
+
```
Comment (babugarnaik) [inline smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java]: Remove extra lines

File: smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+	private String name;
+	
+	@Size(min=2,message="Name should be atleast 2 letters")
+	private String type;
+	
+	
+	private int contactId;
+	private int userId;
+	private boolean byAdmin;
+
```
Comment (babugarnaik) [inline smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java:23]: Remove extra lines

File: smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java [hunk 15]
```diff
@@ -0,0 +1,32 @@
+package com.scm.smartcontactmanager.controller.category.request;
+
+import jakarta.validation.constraints.NotBlank;
```
Comment (babugarnaik) [inline smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java]: Are you using this import?

File: smartcontactmanager/src/main/java/com/scm/smartcontactmanager/entity/category/Category.java [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+	@Id
+	private int id;
+	
+	@Column(name="name")
+	private String name;
+	
+	@Column(name="type")
+	private String type;
+	
+	@Column(name="contat_id")
```
Comment (babugarnaik) [inline smartcontactmanager/src/main/java/com/scm/smartcontactmanager/entity/category/Category.java]: Is column name correct?

**Review by babugarnaik — CHANGES_REQUESTED** [review]

File: Common/yaml/category.yaml (hunk 1 above)
Comment (Subhojitgh) [reply Common/yaml/category.yaml]: I have fixed it. Please review again.

File: Common/yaml/category.yaml (hunk 2 above)
Comment (Subhojitgh) [reply Common/yaml/category.yaml]: I have fixed it. Please review again.

File: Common/yaml/category.yaml (hunk 3 above)
Comment (Subhojitgh) [reply Common/yaml/category.yaml]: I have fixed it. Please review again.

File: Common/yaml/category.yaml (hunk 4 above)
Comment (Subhojitgh) [reply Common/yaml/category.yaml]: I have fixed it. Please review again.

File: Common/yaml/category.yaml (hunk 5 above)
Comment (Subhojitgh) [reply Common/yaml/category.yaml]: I have fixed it. Please review again.

File: Common/yaml/category.yaml (hunk 12 above)
Comment (Subhojitgh) [reply Common/yaml/category.yaml]: I have fixed it. Please review again.

File: smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java (hunk 13 above)
Comment (Subhojitgh) [reply smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java]: I have fixed it. Please review again.

File: smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java (hunk 14 above)
Comment (Subhojitgh) [reply smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java:23]: I have fixed it. Please review again.

File: smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java (hunk 15 above)
Comment (Subhojitgh) [reply smartcontactmanager/src/main/java/com/scm/smartcontactmanager/controller/category/request/CategoryRequestDto.java]: I have fixed it. Please review again.

File: smartcontactmanager/src/main/java/com/scm/smartcontactmanager/entity/category/Category.java (hunk 16 above)
Comment (Subhojitgh) [reply smartcontactmanager/src/main/java/com/scm/smartcontactmanager/entity/category/Category.java]: I have fixed it. Please review again.

**Review by babugarnaik — CHANGES_REQUESTED** [review]
Changes still not done. Please check and submit again

**Comment by Subhojitgh** [comment]
> Changes still not done. Please check and submit again

I have fixed it. Please review again

**Review by babugarnaik — APPROVED** [review]

**Pull request merged**